### ISSUE TITLE
Word generation using LSTM

### ISSUE THREAD

**Issue by kaushalshetty** [keras-team/keras#7677]
Hi all,
I have built a LSTM model that spits out the next word given a sequence of words . But the network is spitting out only the word 'the' as the next predicted word. After that i checked out frequency distribution of the labels where I found out that stopwords were constituting a large number. I can remove the stopwords but if i do so then the final generated results would not make any sense with the absence of stopwords. This is like a classical skewed data problem and how do i get about this ? Would random sampling from the highest probabilities make any sense ? Please help.

**Comment by samre12** [comment]
@kaushalshetty , random sampling will also be prone to the same issue when considered over long term. Its better to train your LSTM using **weighted softmax** classifier in order to account for skewed distribution in your training dataset.

**Comment by kaushalshetty** [comment]
@samre12 The predicted words were better when I implemented the same using a one hot encoded version as input rather than a sequence of indices. Using the concept of diversity(temperature) further diversified my generated words. 
 Thanks

**Comment by samre12** [comment]
@kaushalshetty are the number of words in your vocabulary fixed? Another alternative could to use word models such as *word2vec* or *Glove* etc. to get word representations. Using these as input to the LSTM could even further improve the performance of your model.

**Comment by kaushalshetty** [comment]
@samre12 Sure that would be a great approach and I suppose it will give better results also.
Thanks.
